src/ClientRuntime/Microsoft.Rest.ClientRuntime.E2E.Tests/ScenarioTests/E2ETestBase.cs

```
@@ -0,0 +1,793 @@
+                    };
+
+                    storageAccountOutput = StorageClient.StorageAccounts.Create(rgName,
+                        storageAccountName, stoInput);
+                    bool created = false;
+                    while (!created)
+                    {
+                        System.Threading.Thread.Sleep(TimeSpan.FromSeconds(10));
+                        var stos = StorageClient.StorageAccounts.ListByResourceGroup(rgName);
+                        created =
+                            stos.Any(
+                                t =>
+                                    StringComparer.OrdinalIgnoreCase.Equals(t.Name, storageAccountName));
+                    }
+
+                    storageAccountOutput = StorageClient.StorageAccounts.GetProperties(rgName, storageAccountName);
+                }
+
+                return storageAccountOutput;
+            }
+            catch
+            {
+                //ResourceClient.ResourceGroups.Delete(rgName);
+                throw;
+            }
+        }
+
+        protected ResourceGroup CreateResourceGroup(string rgName)
+        {
+            ResourceGroup resourceGroup = null;
+            try
+            {
+                resourceGroup = ResourceClient.ResourceGroups.Get(rgName);
+                //ResourceClient.ResourceGroups.Delete(rgName);
+            }
+            catch { }
+
+            try
+            {
+                if (resourceGroup == null)
+                {
+                    resourceGroup = ResourceClient.ResourceGroups.CreateOrUpdate(
+                                 rgName,
+                                 new ResourceGroup
+                                {
+                                    Location = DEFAULT_LOCATION,
+                                    Tags = new Dictionary<string, string>() { { rgName, DateTime.UtcNow.ToString("u") } }
+                                });
+                }
+            }
+            catch
```

why catch the exception to throw it?  Just let the exception bubble up